### DISCUSSION TITLE
How to run locally?

### DISCUSSION THREAD

**Discussion by Anu-123-gif** [Project-Books/book-project#721] [Q&A]
I tried to create an account but it wasn't possible when running on localhost. I directly went to [http://localhost:3000/my-books](http://localhost:3000/my-books) and I could a sample view of the website. I'm not sure this is how I'm supposed to run it on my pc.
Please guide me. Thanks !

**Comment by knjk04** [comment]
Is the backend running? 

To create an account when the frontend is running, you'll need to follow all of the steps here: https://github.com/Project-Books/book-project/tree/0.2.0#setup. 

Let us know if the problem still persists

**Reply by Anu-123-gif** [reply]
I used `sudo docker-compose up -d mysql phpmyadmin` to run the backend,
and `yarn start` to run the frontend.
Then i tried to sign up at [http://localhost:3000/sign-up](url) , but nothing happened on clicking 'create account'.
Do I need to do something else as well?

Thanks !

**Reply by knjk04** [reply]
> I used sudo docker-compose up -d mysql phpmyadmin to run the backend,

This runs the database. You'll also need to run the backend. You can either do this from an IDE, like IntelliJ, or by running `java -jar target/book-project-0.2.0.jar` (you'll need to have JDK 11+ installed and navigate to the root of the project)

**Reply by Anu-123-gif** [reply]
Hey,
After following the steps above I was able to sign in using the test user credentials, but I was still not able to create a new account :').
For the time being i'll start working with the test user.

**Reply by knjk04** [reply]
Hi, did you get an error on the backend?

**Reply by Anu-123-gif** [reply]
No I didn't get any error
